src/agent/library/world.js

```diff
@@ -39,6 +39,77 @@ export function getNearestFreeSpace(bot, size=1, distance=8) {
 }
 
 
+export function getBlockAtPosition(bot, x=0, y=0, z=0) {
+     /**
+     * Get a block from the bot's relative position 
+     * @param {Bot} bot - The bot to get the block for.
+     * @param {number} x - The relative x offset to serach, default 0.
+     * @param {number} y - The relative y offset to serach, default 0.
+     * @param {number} y - The relative z offset to serach, default 0. 
+     * @returns {Block} - The nearest block.
+     * @example
+     * let blockBelow = world.getBlockAtPosition(bot, 0, -1, 0);
+     * let blockAbove = world.getBlockAtPosition(bot, 0, 2, 0); since minecraft position is at the feet
+     **/
+    let block = bot.blockAt(bot.entity.position.offset(x, y, z));
+    if (!block) block = {name: 'air'};
+       
+    return block;
+}
+
+
+export function getLowestBlock(bot, block_types=null, ignore_types=null, distance=32) {
```

Comment:
- rename this function to getFirstBlockAboveHead
- block_types is not necessary, only keep the ignore_types param
- ignore cave_air too